src/tests/resistance/fuzzing_corrupted_messages.rs

```diff
@@ -169,47 +192,53 @@ async fn corrupted_version_inplace_of_verack_when_node_initiates_handshake() {
 
     let locked_rng = Arc::new(RwLock::new(seeded_rng()));
 
-    // create tcp listeners for peer set (port is only assigned on tcp bind)
-    let mut listeners = Vec::with_capacity(ITERATIONS);
+    // create peers (we need their ports to give to the node)
+    let mut peers = Vec::with_capacity(ITERATIONS);
     for _ in 0..ITERATIONS {
-        listeners.push(TcpListener::bind(new_local_addr()).await.unwrap());
+        let peer = SyntheticNode::builder()
+            .with_version_exchange_handshake()
+            .with_all_auto_reply()
+            .build()
+            .await
+            .unwrap();
+
+        peers.push(peer);
     }
 
     // get list of peer addresses to pass to node
-    let peer_addresses = listeners
+    let peer_addresses = peers
         .iter()
-        .map(|listener| listener.local_addr().unwrap())
+        .map(|peer| peer.listening_addr())
         .collect::<Vec<_>>();
 
     // start peer processes
-    let mut peer_handles = Vec::with_capacity(listeners.len());
-    for peer in listeners {
+    let mut peer_handles = Vec::with_capacity(peers.len());
+    for mut peer in peers {
         let peer_rng = locked_rng.clone();
         peer_handles.push(tokio::time::timeout(
             tokio::time::Duration::from_secs(120),
             tokio::spawn(async move {
-                // Await connection and receive version
-                let (mut peer_stream, _) = peer.accept().await.unwrap();
-                let version = Message::read_from_stream(&mut peer_stream).await.unwrap();
-                assert_matches!(version, Message::Version(..));
+                // Await connection
+                let node_addr = peer.wait_for_connection().await;
 
-                // send version, receive verack
-                let version = Message::Version(Version::new(
-                    peer_stream.peer_addr().unwrap(),
-                    peer_stream.local_addr().unwrap(),
-                ));
-                version.write_to_stream(&mut peer_stream).await.unwrap();
-                let verack = Message::read_from_stream(&mut peer_stream).await.unwrap();
+                // Receive verack
+                let (_, verack) = peer.recv_message().await;
                 assert_matches!(verack, Message::Verack);
 
-                // send bad version instead of verack
+                // Send bad version instead of verack
+                let version = Message::Version(Version::new(node_addr, peer.listening_addr()));
```

Comment:
I think this naming is slightly confusing, perhaps `node` and `synthetic_node`/`synth_node`/ `s_node` would be better? (I think we should actually choose a convention we stick with throughout the codebase).

Reply:
Yeah good idea. `synth_node` and `node` sound good to me?